### PR TITLE
[5.8] fix MorphTo Relation ignores parent $timestamp when touching 

### PR DESCRIPTION
**Pull request by mx2s**
Fixed bug with touching model which has $timestamps set to false resulting in Illuminate\Database\QueryException: SQLSTATE[HY000]: General error: 1 no such column: updated_at

for source issue and more details see #28638

### CODE REVIEW

**Comment by mxss** [comment]
@driesvints reopened #28658

**Comment by staudenmeir** [comment]
We also have to cover cases where timestamps are enabled, but `UPDATED_AT` is `null`.

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -294,6 +294,10 @@ public static function isIgnoringTouch($class = null)
     {
         $class = $class ?: static::class;
 
+        if (! get_class_vars($class)['timestamps']) {
```
Comment (staudenmeir) [inline src/Illuminate/Database/Eloquent/Model.php]: I would prefer `(new $class)->usesTimestamps()` to be consistent with existing code.

Comment (mxss) [reply src/Illuminate/Database/Eloquent/Model.php]: > We also have to cover cases where timestamps are enabled, but UPDATED_AT is null:
just added test

Comment (mxss) [reply src/Illuminate/Database/Eloquent/Model.php]: @staudenmeir in some cases we're getting abstract classes here for ex.
https://github.com/laravel/framework/blob/5992f10116f612d51465385b0af11d778d975b87/tests/Database/DatabaseEloquentIntegrationTest.php#L1539

Comment (staudenmeir) [reply src/Illuminate/Database/Eloquent/Model.php]: Yes, but only in the tests. You can't get an abstract class in a real application. Please adjust the test accordingly.

File: tests/Database/DatabaseEloquentModelTest.php [hunk 2]
```diff
@@ -1896,6 +1896,27 @@ protected function addMockConnection($model)
...
+    public function testTouchingModelWithTimestamps()
+    {
+        $this->assertFalse(
+            Model::isIgnoringTouch(Model::class)
+        );
+    }
+
+    public function testTouchingModelWithUpdatedAtNull()
+    {
+        $this->assertFalse(
```
Comment (staudenmeir) [inline tests/Database/DatabaseEloquentModelTest.php]: This case has to return `true`. You need to cover it by adjusting `isIgnoringTouch()`.

File: src/Illuminate/Database/Eloquent/Model.php (hunk 1 above)
Comment (mxss) [reply src/Illuminate/Database/Eloquent/Model.php]: > Yes, but only in the tests. You can't get an abstract class in a real application.

OK, but If I use `(new $class)->usesTimestamps()` 6 more test are failing (not including mine) - what to do with that?
For example this one:
https://github.com/laravel/framework/blob/5992f10116f612d51465385b0af11d778d975b87/tests/Database/DatabaseEloquentIntegrationTest.php#L1539

File: tests/Database/DatabaseEloquentModelTest.php (hunk 2 above)
Comment (mxss) [reply tests/Database/DatabaseEloquentModelTest.php]: done

File: src/Illuminate/Database/Eloquent/Model.php (hunk 1 above)
Comment (staudenmeir) [reply src/Illuminate/Database/Eloquent/Model.php]: I think we should move the new code to `Relation::touch()`. `Model::isIgnoringTouch()` is only about `$ignoreOnTouch`, not about timestamps.

**Pull request merged**